llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -3253,12 +3253,13 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
     // call void @llvm.assume(i1 %D)
     // into
     // call void @llvm.assume(i1 true) [ "align"(i32* [[A]], i64  Constant + 1)]
-    uint64_t AlignMask;
+    uint64_t AlignMask = 1;
     if (EnableKnowledgeRetention &&
-        match(IIOperand,
-              m_SpecificICmp(ICmpInst::ICMP_EQ,
-                             m_And(m_Value(A), m_ConstantInt(AlignMask)),
-                             m_Zero()))) {
+        (match(IIOperand,
+               m_SpecificICmp(ICmpInst::ICMP_EQ,
+                              m_And(m_Value(A), m_ConstantInt(AlignMask)),
+                              m_Zero())) ||
+         match(IIOperand, m_Not(m_Trunc(m_Value(A)))))) {
```

Comment:
I think `trunc` match needs to be first. A failed pattern match is not guranteed to not modify the values.

Reply:
Can change the order but it is not possible to make a test that will fail as the m_and sub pattern will not execute unless it is an icmp instruction and in that case it can not also match the n_not pattern.

Reply:
In this case I think you're right that it won't be an issue either way.

Reply:
fixed